lib/common/client/base.rb

```diff
@@ -53,10 +54,13 @@ def perform(method, path, params, headers = nil)
         send(method, path, params || {}, headers || {})
       end
 
+      # rubocop:disable Metrics/MethodLength
       def request(method, path, params = {}, headers = {})
         sanitize_headers!(method, path, params, headers)
         raise_not_authenticated if headers.keys.include?('Token') && headers['Token'].nil?
-        connection.send(method.to_sym, path, params) { |request| request.headers.update(headers) }.env
+        handle_service_unavailable do
+          connection.send(method.to_sym, path, params) { |request| request.headers.update(headers) }
+        end.env
```

Comment:
can't this live inside application_controller or maybe a processor? I'm not sure I agree with this change for all services. For example, MHV is going to encounter 503 very rarely. In the rare event that it does arise, I don't really see any problem with it being a normal exception (vs warning). Furthermore, I don't like making these types of changes in the client. Pretend the client is a gem, that you can't modify... handling these one-off error issues is not something it should be concerned with.